an-apluss/teamwork
# 28421130 Implement the functionality for an employee to create gif post

#### What does this PR do?
- Implement the functionality for an employee to create gif post

#### Description of Task to be completed?
- write the test suite for the employee to create gifs post
- add logic to create a new gif image post
- add logic to validate user input
- make failing test to pass
- add Cloudinary environment variable to .env.example file
- finish[ #7 ]

#### How should this be manually tested?
- Clone the repository using git clone `https://github.com/an-apluss/teamwork.git`  in your terminal
- run `npm install` on your terminal
- run `npm run migrations` on your terminal to create required tables for the application
- run `npm start` on your terminal to start the server
- make a `post` request on `https://web-teamwork.herokuapp.com/api/v1/gifs` with the necessary body request using postman
- ensure you set the request header like below
````
key                                Value
Authorization              Bearer wwedncndhd.ckd.--cmdmksks...
````
- use the below body request data
````
key                                Value
title                       thank God its Friday
image                  c://users/programs/images/weekend.gif
````
- you should see a response like below 
````
{
   "status": "success",
   "data": {
         "gifId": 2,
        "message": "GIF image successfully posted",
        "createdOn": "2019/11/07 13:05",
        "title": thank God is friday,
        "imageUrl": "https://res.cloudinary.com.."
   }
}
````
#### Any background context you want to provide?
Ensure that you provide value for the list of environment variable in the `.env.example`  file and rename to `.env`

#### What are the relevant issue stories?
- Github Project Board Card Link - ( [#28421130](https://github.com/an-apluss/teamwork/projects/1#card-28421130) )

#### Screenshots(if appropriate)
-N/A
